### PR TITLE
show info of route conflict

### PR DESCRIPTION
**Pull request by pursehouse**
This patch updates the exception thrown when a route conflict occurs to include information about the path and HTTP methods of the route causing the conflict.

### CODE REVIEW

File: src/RouteCollector.php [hunk 1]
```diff
@@ -164,8 +164,10 @@ private function checkForDuplicateRoute(string $path, array $methods = null) : v
         });
 
         if (! empty($matches)) {
+            $first = reset( $matches );
+            $example = implode( ",", $first->getAllowedMethods() ?: [] ). " : ".$first->getPath(). " : ".$first->getName();
             throw new Exception\DuplicateRouteException(
-                'Duplicate route detected; same name or path, and one or more HTTP methods intersect'
+                sprintf('Duplicate route detected; "%s"', $example)
```
Comment (xtreamwayz) [inline src/RouteCollector.php]: Use something like this please:
```php
if (! empty($matches)) {
    $match = reset($matches);
    throw new Exception\DuplicateRouteException(sprintf(
        'Duplicate route detected; "%s : %s : %s"',
        implode(',', $match->getAllowedMethods() ?? []),
        $match->getPath(),
        $match->getName()
    ));
}
```

File: src/RouteCollector.php [hunk 2]
```diff
@@ -164,9 +164,13 @@ private function checkForDuplicateRoute(string $path, array $methods = null) : v
         });
 
         if (! empty($matches)) {
-            throw new Exception\DuplicateRouteException(
-                'Duplicate route detected; same name or path, and one or more HTTP methods intersect'
-            );
+            $match = reset( $matches );
```
Comment (xtreamwayz) [inline src/RouteCollector.php]: 167 | ERROR | [x] Space after opening parenthesis of function call prohibited
 167 | ERROR | [x] Expected 0 spaces before closing bracket; 1 found

Remove the space around `$matches`

File: src/RouteCollector.php [hunk 3]
```diff
@@ -164,9 +164,13 @@ private function checkForDuplicateRoute(string $path, array $methods = null) : v
...
+            throw new Exception\DuplicateRouteException(sprintf(
+                'Duplicate route detected; "%s : %s : %s"',
+                implode( ',', $match->getAllowedMethods() ?? [] ),
```
Comment (xtreamwayz) [inline src/RouteCollector.php]: 170 | ERROR | [x] Space after opening parenthesis of function call prohibited
 170 | ERROR | [x] Expected 0 spaces before closing bracket; 1 found

Same here, no spaces issues.

File: src/RouteCollector.php [hunk 4]
```diff
@@ -164,9 +164,13 @@ private function checkForDuplicateRoute(string $path, array $methods = null) : v
         });
 
         if (! empty($matches)) {
-            throw new Exception\DuplicateRouteException(
-                'Duplicate route detected; same name or path, and one or more HTTP methods intersect'
-            );
+            $match = reset($matches);
+            throw new Exception\DuplicateRouteException(sprintf(
+                'Duplicate route detected; "%s : %s : %s"',
+                implode(',', $match->getAllowedMethods() ?? []),
```
Comment (webimpress) [inline src/RouteCollector.php]: I would have here `(any)` instead of empty array

**Comment by weierophinney** [comment]
Thanks, @pursehouse

**Pull request merged**